lib/streamlit/runtime/caching/cache_utils.py

```diff
@@ -92,6 +151,35 @@ class CachedResult:
     sidebar_id: str
 
 
+@dataclass
+class WidgetStateNotFound:
+    pass
+
+
+WIDGET_STATE_NOT_FOUND = WidgetStateNotFound()
```

Comment:
I wanted it to be something that will never be a valid widget state value. It would be weird for someone to have a widget return an empty object instance, but not impossible, while one returning this sentinel would require actively trying to fuck with our internals